x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rules.ts

```diff
@@ -116,7 +114,7 @@ export function useLoadRules({
         http,
         page,
         searchText,
-        typesFilter: typesFilter && typesFilter.length > 0 ? typesFilter : filteredRuleTypes,
+        typesFilter,
```

Comment:
@XavierM I discussed about this with @maryam-saeidi already and by removing this we end up not showing at all the `noData` screen in o11y rules page.

**No data screen in Stack Management** 
<img width="1399" alt="Screenshot 2022-07-08 at 22 52 03" src="https://user-images.githubusercontent.com/2852703/178068960-9829cdb1-22e5-459f-b099-e37b3da590d5.png">


`Screen in o11y rules page` after removing above code
<img width="1402" alt="Screenshot 2022-07-08 at 22 53 26" src="https://user-images.githubusercontent.com/2852703/178069066-c1423dcd-467b-4eba-836f-1d520f66f935.png">

The reason this is broken, is that for o11y rules page [this line](https://github.com/elastic/kibana/blob/main/x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rules.ts#L140) is false on first rendering (user might not have filtered to see a specific rule type from the Type dropdown, but we (o11y) by default filter only the o11y rule types). That's why I ended up passing an extra prop `filteredRuleTypes` to differentiate the user selected filter with the default filter on rendering. Hope that makes sense. I had it as you have it now, but I spotted the `No data screen` bug and had to do it with the extra prop. If you find a better way to fix, go ahead. But right now it is broken. In o11y we also need to show the noData screen.

cc @maryam-saeidi